### PR TITLE
FIX - updating file access to supported mode in >= Marshmallow 

### PR DESCRIPTION
**Pull request by rksh**
MODE_WORLD_READABLE is depr. in M, fails in N.

### CODE REVIEW

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MainActivity.java [hunk 1]
```diff
@@ -1561,7 +1568,7 @@ public void playRunNetSound() {
 
     private void setupWifi() {
         // warn about turning off network notification
-        @SuppressWarnings("deprecation")
+        //@SuppressWarnings("deprecation")
```
Comment (bobzilladev) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MainActivity.java]: can pry delete

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MainActivity.java [hunk 2]
```diff
@@ -894,8 +894,11 @@ public static FileOutputStream createFile(final Context context, final String fi
             return new FileOutputStream(file);
         }
 
-        //noinspection deprecation
-        return context.openFileOutput(filename, Context.MODE_WORLD_READABLE);
+        if (Build.VERSION.SDK_INT >= 23) {
+            return context.openFileOutput(filename, Context.MODE_PRIVATE);
+        } else {
+            return context.openFileOutput(filename, Context.MODE_WORLD_READABLE);
+        }
```
Comment (bobzilladev) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MainActivity.java]: I'd do a ternary to chose the mode to a variable, then pass that variable to a single openFileOutput, to reduce copy paste. i don't know if android developer inspection handles that properly though.

Comment (rksh) [reply wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MainActivity.java]: yeah, i was satisfying the noodly editor validation logic. i'll read up on the right way to do it.

**Comment by bobzilladev** [comment]
as long as this doesn't break upgrades over existing files, and can still get to sd files, i'm +100

**Pull request merged**